### ISSUE TITLE
media_player status will not update after wifi drop

### ISSUE THREAD

**Issue by walthowd** [custom-components/alexa_media_player#406]
**Describe the bug**
If WiFi drops on the HA server and the Echo Dot while media is playing then the media player status (playing, idle, standby) will be stuck at the current state and any subsequent operations to change media, turn off, etc will not work. 

We try to play white noise each night, a playlist from Amazon Music, but each night sometime between 4AM and 6AM our WiFi router was rebooting to try to apply an update. 

The hardware we are using is a 2nd generation Echo Dot with serial ending in EMQ. 

Here is a screenshot of the media_player status. The router reboots at 6:13AM and the status sticks as playing and ignores any future operations

**To Reproduce**
1. Create a long white noise playlist on Amazon Music
2. Call play_media for that playlist in HA
3. Reboot WiFi router. 

**Expected behavior**
alexa_media_player may try to re-establish connection or reinitialize after a disconnect. 

**Screenshots**
Media Player status:
https://www.dropbox.com/s/z3hs1q3f0z89zxx/Screenshot%202019-09-30%2009.02.16.png?dl=0

Playlist screenshot:
https://www.dropbox.com/s/0howjk2angmv23s/Screenshot%202019-09-30%2009.13.48.png?dl=0

**System details**
 - Home-assistant (version):  0.99.2
 - Hassio (Yes/No): No
 - alexa_media  (version from `const.py` or HA startup): 2.3.0
 - alexapy (version from `pip show alexapy` or HA startup): 1.3.1

**Logs**
[alexa_media_logs.txt](https://github.com/custom-components/alexa_media_player/files/3671874/alexa_media_logs.txt)

**Comment by alandtse** [comment]
Work around would be to use config entries and remove then add. Let me review the logs to see if there's an event to catch to trigger the reconnect.

**Comment by alandtse** [comment]
Likely the same issue as #397

**Comment by walthowd** [comment]
Very easy to replicate for me, agree that the other bug covers the options that still work after losing the socket.

**Comment by alandtse** [comment]
Please try 2.3.1

**Comment by Dielee** [comment]
Not working for me, sorry

**Comment by Dielee** [comment]
![ha](https://user-images.githubusercontent.com/45101649/66462294-0fe42e00-ea7b-11e9-8d7c-d84e16d68593.gif)

**Comment by funkschrauber** [comment]
I can confirm that. Still no status update. Restarted two times.

**Comment by Dielee** [comment]
@funkschrauber can you reopen the issue ?

**Comment by walthowd** [comment]
Updated to 2.3.1 today and received this error when I was about to re-test:

```
2019-10-09 15:58:30 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection.139875941900688] 'configurator'
Traceback (most recent call last):
  File "/config/custom_components/alexa_media/helpers.py", line 143, in wrapper
    result = await func(*args, **kwargs)
  File "/config/custom_components/alexa_media/media_player.py", line 796, in async_play_media
    customer_id=self._customer_id, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/alexapy/alexaapi.py", line 291, in play_music
    musicProviderId=provider_id)
  File "/usr/local/lib/python3.7/site-packages/alexapy/alexaapi.py", line 208, in send_sequence
    data=data)
  File "/usr/local/lib/python3.7/site-packages/alexapy/alexaapi.py", line 90, in _post_request
    return await self._request('post', uri, data)
  File "/usr/local/lib/python3.7/site-packages/alexapy/alexaapi.py", line 82, in _request
    raise AlexapyLoginError(response.reason)
alexapy.errors.AlexapyLoginError: Unauthorized

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 130, in handle_call_service
    connection.context(msg),
  File "/usr/src/homeassistant/homeassistant/core.py", line 1234, in async_call
    await asyncio.shield(self._execute_service(handler, service_call))
  File "/usr/src/homeassistant/homeassistant/core.py", line 1259, in _execute_service
    await handler.func(service_call)
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 213, in handle_service
    self._platforms.values(), func, call, service_name, required_features
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 349, in entity_service_call
    future.result()  # pop exception if have
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 373, in _handle_service_platform_call
    await func(entity, data)
  File "/config/custom_components/alexa_media/helpers.py", line 159, in wrapper
    ['configurator'])):
KeyError: 'configurator'
```

**Comment by alandtse** [comment]
@walthowd please see #423 if you have the same errors in 2.3.1.  From the logs, you are having authorization issues with your current pickle so you should rename it or delete it. Otherwise you're probably experiencing degraded functionality and not realizing it.

**Comment by walthowd** [comment]
@alandtse I agree it's odd -- but I can confirm I'm running now for 2 days with the same pickle and what appears to be full functionality with 2.3.0 -- All media player operations and states work and are correct. 

I just now upgraded to 0.100.1 HA and back to 2.3.1 and the same pickle seems happy now. 

Tested again by rebooting wifi router:

Started Amazon music stream on media_player.colins_room

11:02 local time -Rebooted router
11:05 Turned off media player (reported success)
11:05 to 11:07 - Several times tried to start media player playing amazon music playlist - Cover art instantly updated but media_player state stayed "paused".
11:07 - Third try of media_player.play_media the media_player state updated to "Playing"

Logs attached. 

[alexa_media_logs.txt](https://github.com/custom-components/alexa_media_player/files/3718604/alexa_media_logs.txt)

**Comment by walthowd** [comment]
@alandtse  ISP went down this morning at 5:07AM, came back at 5:38. Looks like alexa_media noticed and retried four times to connect, then gave up for good. Logs:

```
2019-10-16 05:25:08 DEBUG (MainThread) [custom_components.alexa_media] w******d@g*******m: Received websocket error #0 <class 'concurrent.futures._base.TimeoutError'>
2019-10-16 05:25:08 DEBUG (MainThread) [custom_components.alexa_media] w******d@g*******m: Websocket closed; reconnect #1 in 10s
2019-10-16 05:25:18 DEBUG (MainThread) [custom_components.alexa_media] w******d@g*******m: Websocket created: <alexapy.alexawebsocket.WebsocketEchoClient object at 0x7fa79d770390>
2019-10-16 05:25:23 DEBUG (MainThread) [custom_components.alexa_media] w******d@g*******m: Websocket creation failed: Cannot connect to host dp-gw-na-js.amazon.com:443 ssl:<ssl.SSLContext object at 0x7fa7bcc48ec0> [Try again]
2019-10-16 05:25:23 DEBUG (MainThread) [custom_components.alexa_media] w******d@g*******m: Websocket closed; reconnect #2 in 20s
2019-10-16 05:25:43 DEBUG (MainThread) [custom_components.alexa_media] w******d@g*******m: Websocket created: <alexapy.alexawebsocket.WebsocketEchoClient object at 0x7fa797549850>
2019-10-16 05:25:48 DEBUG (MainThread) [custom_components.alexa_media] w******d@g*******m: Websocket creation failed: Cannot connect to host dp-gw-na-js.amazon.com:443 ssl:<ssl.SSLContext object at 0x7fa7bcc48ec0> [Try again]
2019-10-16 05:25:48 DEBUG (MainThread) [custom_components.alexa_media] w******d@g*******m: Websocket closed; reconnect #3 in 40s
2019-10-16 05:26:28 DEBUG (MainThread) [custom_components.alexa_media] w******d@g*******m: Websocket created: <alexapy.alexawebsocket.WebsocketEchoClient object at 0x7fa798831710>
2019-10-16 05:26:33 DEBUG (MainThread) [custom_components.alexa_media] w******d@g*******m: Websocket creation failed: Cannot connect to host dp-gw-na-js.amazon.com:443 ssl:<ssl.SSLContext object at 0x7fa7bcc48ec0> [Try again]
2019-10-16 05:26:33 DEBUG (MainThread) [custom_components.alexa_media] w******d@g*******m: Websocket closed; reconnect #4 in 80s
2019-10-16 05:27:53 DEBUG (MainThread) [custom_components.alexa_media] w******d@g*******m: Websocket created: <alexapy.alexawebsocket.WebsocketEchoClient object at 0x7fa79780d4d0>
2019-10-16 05:27:58 DEBUG (MainThread) [custom_components.alexa_media] w******d@g*******m: Websocket creation failed: Cannot connect to host dp-gw-na-js.amazon.com:443 ssl:<ssl.SSLContext object at 0x7fa7bcc48ec0> [Try again]
2019-10-16 05:27:58 DEBUG (MainThread) [custom_components.alexa_media] w******d@g*******m: Websocket closed; retries exceeded; polling
```

Should we keep bumping out the retry time and periodically retrying to connect? As of now it appears to have given up for good and will require a HA restart to bring the connection back.
